src/OrchardCore.Modules/OrchardCore.Lists/Helpers/ListQueryHelpers.cs

```diff
@@ -20,6 +21,7 @@ internal static Task<IEnumerable<ContentItem>> QueryListItemsAsync(ISession sess
     {
         return session.Query<ContentItem>()
                 .With<ContainedPartIndex>(x => x.ListContentItemId == listContentItemId)
+                .OrderBy(o => o.Order)
```

Comment:
You should also add `.ThenBy(o => o.Id)` to ensure some items won't be overlooked.